File: Celestial-Almanac/src/main/cpp/sample_main.cpp
#include <iostream>
#include <stdio.h>
#include <string.h>

#include "AstroComputer.h"

// #include "CelestStruct.h"

const double DELTA_T = 69.2201;

int main () {
  std::cout << "Sample main, calculating for 2020-Mar-28 16:50:20\n";
  // 2020-MAR-28 16:50:20 UTC
  
  double deltaT = DELTA_T;
  // Recalculate deltaT
  deltaT = calculateDeltaT(2020, 3);
  fprintf(stdout, "Setting DeltaT to %f\n", deltaT);

  ComputedData * data = calculate(2020, 3, 28, 16, 50, 20, deltaT);
  fprintf(stdout, "--- Calculated 2020-Mar-28 16:50:20 ---\n");
  fprintf(stdout, "Julian Dates %f %f %f\n", data->JD0h, data->JD, data->JDE);
  fprintf(stdout, "Sideral Time %s\n", data->SidTm);
  char eot[32];
  fprintf(stdout, "EoT: %f => %s\n", data->EoT, outEoT(data->EoT, eot));
  char gha[32], ra[32], dec[32], sd[32], hp[32];
  fprintf(stdout, "---------------- Bodies ---------------\n");
  fprintf(stdout, "Sun \tGHA: %s, RA: %s, Dec: %s, sd: %s, hp: %s\n",
      outHA(data->GHASun, gha),
      outRA(data->RASun, ra),
      outDec(data->DECSun, dec),
      outSdHp(data->SDSun, sd),
      outSdHp(data->HPSun, hp));
  fprintf(stdout, "Venus \tGHA: %s, RA: %s, Dec: %s, sd: %s, hp: %s\n",
      outHA(data->GHAVenus, gha),
      outRA(data->RAVenus, ra),
      outDec(data->DECVenus, dec),
      outSdHp(data->SDVenus, sd),
      outSdHp(data->HPVenus, hp));
  fprintf(stdout, "Mars \tGHA: %s, RA: %s, Dec: %s, sd: %s, hp: %s\n",
      outHA(data->GHAMars, gha),
      outRA(data->RAMars, ra),
      outDec(data->DECMars, dec),
      outSdHp(data->SDMars, sd),
      outSdHp(data->HPMars, hp));
  fprintf(stdout, "Jupiter\tGHA: %s, RA: %s, Dec: %s, sd: %s, hp: %s\n",
      outHA(data->GHAJupiter, gha),
      outRA(data->RAJupiter, ra),
      outDec(data->DECJupiter, dec),
      outSdHp(data->SDJupiter, sd),
      outSdHp(data->HPJupiter, hp));
  fprintf(stdout, "Saturn \tGHA: %s, RA: %s, Dec: %s, sd: %s, hp: %s\n",
      outHA(data->GHASaturn, gha),
      outRA(data->RASaturn, ra),
      outDec(data->DECSaturn, dec),
      outSdHp(data->SDSaturn, sd),
      outSdHp(data->HPSaturn, hp));

  fprintf(stdout, "Moon \tGHA: %s, RA: %s, Dec: %s, sd: %s, hp: %s\n",
      outHA(data->GHAMoon, gha),
      outRA(data->RAMoon, ra),
      outDec(data->DECMoon, dec),
      outSdHp(data->SDMoon, sd),
      outSdHp(data->HPMoon, hp));
  fprintf(stdout, "\tMoon phase: %f, %s\n", data->moonPhaseAngle, data->moonPhase);
  fprintf(stdout, "Polaris\tGHA: %s, RA: %s, Dec: %s\n",
      outHA(data->GHAPol, gha),
      outRA(data->RAPol, ra),
      outDec(data->DECPol, dec));
  fprintf(stdout, "Ecliptic obliquity %s, true %s\n", outECL(data->eps0, ra), outECL(data->eps, dec));
  fprintf(stdout, "Lunar Distance %s\n", outHA(data->LDist, gha));
  fprintf(stdout, "Day of Week %s\n", data->DoW);
  fprintf(stdout, "---------------------------------------\n");
  std::cout << "Done with C!\n";
}

File: Celestial-Almanac/src/main/go/Celest/main/SampleMain.go
package main

import (
	"fmt"
	"time"
	"oliv.cc/astro"
)

func makeTimestamp() int64 {
//     return time.Now().UnixNano() / int64(time.Millisecond)  // ms
    return time.Now().UnixNano() / 1000  // micro-s
}
/**
 * This is an example 
 * showing how to use the AstroComputer
 */
func main() {

	fmt.Println("Let's get started!")

    year := 2020
    month := 3
    day := 28
    hour := 16
    minute := 50
    second := 20

    before := makeTimestamp()
	deltaT := astro.CalculateDeltaT(year, month)
// 	fmt.Printf("DeltaT for %v-%v: %v s\n", year, month, deltaT)

	fmt.Printf("Calculating for %v-%v-%v %v:%v:%v\n", year, month, day, hour, minute, second)
	result := astro.Calculate(year, month, day, hour, minute, second, deltaT)
    after := makeTimestamp()
	fmt.Printf("Calculation took : %v \u03bcs\n", (after - before))
	fmt.Printf("Julian Dates %v %v %v\n", result.JD0h, result.JD, result.JDE)
	fmt.Printf("Sideral Time %s\n", result.SidTm)
	fmt.Printf("EoT: %v => %s\n", result.EoT, astro.OutEoT(result.EoT))

	fmt.Println("------------- Bodies --------------")
	fmt.Printf("Sun \tGHA: %s, RA: %s, Dec: %s, sd: %s, hp: %s\n", 
		astro.OutHA(result.GHASun),
		astro.OutRA(result.RASun),
		astro.OutDec(result.DECSun),
		astro.OutSdHp(result.SDSun),
		astro.OutSdHp(result.HPSun))
	fmt.Printf("Venus \tGHA: %s, RA: %s, Dec: %s, sd: %s, hp: %s\n", 
		astro.OutHA(result.GHAVenus),
		astro.OutRA(result.RAVenus),
		astro.OutDec(result.DECVenus),
		astro.OutSdHp(result.SDVenus),
		astro.OutSdHp(result.HPVenus))
	fmt.Printf("Mars \tGHA: %s, RA: %s, Dec: %s, sd: %s, hp: %s\n", 
		astro.OutHA(result.GHAMars),
		astro.OutRA(result.RAMars),
		astro.OutDec(result.DECMars),
		astro.OutSdHp(result.SDMars),
		astro.OutSdHp(result.HPMars))
	fmt.Printf("Jupiter \tGHA: %s, RA: %s, Dec: %s, sd: %s, hp: %s\n", 
		astro.OutHA(result.GHAJupiter),
		astro.OutRA(result.RAJupiter),
		astro.OutDec(result.DECJupiter),
		astro.OutSdHp(result.SDJupiter),
		astro.OutSdHp(result.HPJupiter))
	fmt.Printf("Saturn \tGHA: %s, RA: %s, Dec: %s, sd: %s, hp: %s\n", 
		astro.OutHA(result.GHASaturn),
		astro.OutRA(result.RASaturn),
		astro.OutDec(result.DECSaturn),
		astro.OutSdHp(result.SDSaturn),
		astro.OutSdHp(result.HPSaturn))

	fmt.Printf("Moon \tGHA: %s, RA: %s, Dec: %s, sd: %s, hp: %s\n", 
		astro.OutHA(result.GHAMoon),
		astro.OutRA(result.RAMoon),
		astro.OutDec(result.DECMoon),
		astro.OutSdHp(result.SDMoon),
		astro.OutSdHp(result.HPMoon))
	fmt.Printf("\tMoon Phase: %v, %s\n", result.MoonPhaseAngle, result.MoonPhase)

  fmt.Printf("Polaris\tGHA: %s, RA: %s, Dec: %s\n",
  		astro.OutHA(result.GHAPol),
  		astro.OutRA(result.RAPol),
  		astro.OutDec(result.DECPol))
  fmt.Printf("Ecliptic obliquity %s, true %s\n", astro.OutECL(result.Eps0), astro.OutECL(result.Eps))
  fmt.Printf("Lunar Distance %s\n", astro.OutHA(result.LDist))
  fmt.Printf("Day of Week %s\n", result.DoW)
  fmt.Println("---------------------------------------")
  fmt.Println("Done with Golang!")

}
